### DISCUSSION TITLE
Change the install option to ALWAYS overwrite user menus (and add an option for updating rooms)

### DISCUSSION THREAD

**Discussion by DarrenTAnims** [opentoonz/opentoonz#6453] [Ideas]
### Is your feature request related to a problem? Please describe

On installation of a new version (full or nightly), if there's a new feature that has a menu command, this command is stored in the settings, so you need to have that setting to see the new command. Very often, users install a new version and report errors that they can't see the new feature. So I instruct them to delete their user settings or, more simple, just rename the OpenToonz stuff folder, so they'll get a fresh install. This isn't realistic to tell every user to do.

### Describe the solution you'd like

Although there is an option on install, "Overwrite all setting files in the stuff folder except user's personal settings", I don't think this should include the menus. They should always be updated.
And secondly, maybe even update the room files where a new panel has been added to a room. The user should get that new room. If needed, add a second tickbox, "Update rooms, to include any new panels"

**Comment by flurick** [comment]
I wonder what the history of the Stuff folder is. My guess is that it makes updating the ui  for a larger studio team easier. 

But perhaps the default should be to not create a separate Stuff directory.
Blender for example keeps separate configs for each version. And I think OT can already load configurations from within each “portable” install.

**Comment by DarrenTAnims** [comment]
@shun-iwasawa - If it's not possible to update the installer, can the download page explain to rename or delete the stuff folder before installing? Or to reset the rooms on first opening?

**Comment by gab3d** [comment]
In Blender, when installing a new version of the program for the first time, and if a previous version of the program was used (so configuration files for that version exist in the system).
When you first run the program, a popup emerges alerting the user of such situation, and letting him/her decide if it's better to start with completely new config files, or use the ones of the previous version instead.

![imagen](https://user-images.githubusercontent.com/18104091/220449633-ffe046b9-6149-4a32-b18e-7daecfc7236b.png)


I think that would also be good for OT.

P.S. (under Windows) Blender stores its settings in "C:\Users\<user>\AppData\Roaming\Blender Foundation\Blender\<version_number>\" folder, so when a new version is used, a new "<version_number>"  folder is created in the said settings folder; that way several versions of the software can peacefully coexist.
Independently of where OT settings are stored, I think it could be a good thing that OT also saved its settings files in a subfolder exclusive for each version, as that would alleviate much of the upgrading problems we're constantly seeing.

**Comment by konero** [comment]
Some thoughts on this:
1. Users shouldn't be renaming or deleting their stuff folder, there's no need
2. Always resetting rooms would make it a pain for people who customize their layout and menus

I think the best approach might be to make it so the Default set is non-customizable for the menus and always in-sync, it can just use the hard-coded entries. If you make a new room set, then it can be customizable, which I think is an extremely small number of people who actually do it anyway.

**Comment by manongjohn** [comment]
If I may make a suggestion....

When rooms are created, a room#_menubar.xml file is created for that room automatically.  I would suggest only create that file when the room's menubar is customized and removed if it exactly matches the default. Otherwise, if no menubar definition file is found, use the default.

In this way, when upgrading, if there is a change to the default menus, it will automaically show new/modified stuff. Only customized menubars would not be updated.

Might be useful to also have an option to reset the menubar to the default one.

**Comment by gab3d** [comment]
Note on my previous message:
(under Windows) Blender stores its settings in "C:\Users\<user>\AppData\Roaming\Blender Foundation\Blender\<version_number>\" folder, so when a new version is used, a new "<version_number>"  folder is created in the said settings folder; that way several versions of the software can peacefully coexist.
Independently of where OT settings are stored, I think it could be a good thing that OT also saved its settings files in a subfolder exclusive for each version, as that would alleviate much of the upgrading problems we're constantly seeing.

**Comment by gab3d** [comment]
@manongjohn 
One thing I wouldn't like to lose is the ability to delete default rooms, if they are considered of no use by the user.

I also think that something should be done to the way default menus are handled during upgrades.

**Comment by manongjohn** [comment]
@gab3d my solution doesn't have anything to do with the room itself, just the menubar, as it is defined in a separate file from the room itself. 

Rooms should be deletable if the user doesn't want it. If I gave the impression it shouldn't then I appologize. 

Upgrading OT really shouldn't need to have the stuff folder moved aside. It should be able to upgrade in place without issue. It is extremely annoying to have to reset up everything after an upgrade if you have a ton of customizations.

**Comment by RodneyBaker** [comment]
Converting this to a Discussion for additional exploration.
